app/controllers/application_controller.rb

```diff
@@ -108,6 +108,6 @@ def staff?
   end
 
   def true_user
-    @true_user ||= User.find(session[:user_id]) if session[:user_id]
+    @true_user ||= User.find_by(id: session[:user_id])
```

Comment:
It might be worth adding a small integration test to show we do get redirected. Code LGTM though!